documents/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,1081 @@
+
+```
+builder->WithPublishReceivedCallback(
+        [&](const Mqtt5::PublishReceivedEventData &eventData) {
+            /*Called when a message is received by one of the active subscriptions.*/
+        });
+
+std::shared_ptr<Aws::Crt::Mqtt5::Mqtt5Client> client = builder->Build();
+
+Mqtt5::Subscription sub1("my/own/topic",
+                         Mqtt5::QOS::AWS_MQTT5_QOS_AT_LEAST_ONCE);
+std::shared_ptr<Mqtt5::SubscribePacket> subPacket =
+        std::make_shared<Mqtt5::SubscribePacket>();
+subPacket->WithSubscription(std::move(sub1));
+
+auto onSubAck = [&](int error_code,
+                    std::shared_ptr<Mqtt5::SubAckPacket> suback)
+{ /* acknowledge callback */ };
+
+// Subscribe to topic
+bool rc = client->Subscribe(subPacket, onSubAck);
+
+```
+
+
+
+### Unsubscribe
+
+V1 SDK provides blocking and non-blocking API for unsubscribing. To unsubscribe from topic in V1, you should provide a `std::vector` of `std::unique_ptr` of `Utf8String` to the [Unsubscribe](http://aws-iot-device-sdk-cpp-docs.s3-website-us-east-1.amazonaws.com/latest/classawsiotsdk_1_1_mqtt_client.html#a8dc5fa8e8c1522219e6df33cbaa7e376) operation.
+For asynchronous operations use [UnsubscribeAsync](http://aws-iot-device-sdk-cpp-docs.s3-website-us-east-1.amazonaws.com/latest/classawsiotsdk_1_1_mqtt_client.html#a4577dd3e720dea692755c640d4c638ed)  success and failure results are sent through the callback [ActionData::AsyncAckNotificationHandlerPtr](http://aws-iot-device-sdk-cpp-docs.s3-website-us-east-1.amazonaws.com/latest/classawsiotsdk_1_1_action_data.html#a5d1d7452e081205b414e4df985d82f60) 
+
+V2 SDK provides only asynchronous non-blocking API. First, you need to create an [UnsubscribePacket](https://aws.github.io/aws-iot-device-sdk-cpp-v2/class_aws_1_1_crt_1_1_mqtt5_1_1_unsubscribe_packet.html) object with the help of [UnsubscribePacketBuilder](https://awslabs.github.io/aws-crt-java/software/amazon/awssdk/crt/mqtt5/packets/UnsubscribePacket.UnsubscribePacketBuilder.html). The [Unsubscribe](https://aws.github.io/aws-iot-device-sdk-cpp-v2/class_aws_1_1_crt_1_1_mqtt5_1_1_mqtt5_client.html#a96a931b49893d54712062722c5ab7d1a) operation takes a description of the [UnsubscribePacket](https://aws.github.io/aws-iot-device-sdk-cpp-v2/class_aws_1_1_crt_1_1_mqtt5_1_1_unsubscribe_packet.html) you wish to send and returns a promise that resolves successfully with the corresponding [UnsubAckPacket](https://aws.github.io/aws-iot-device-sdk-cpp-v2/class_aws_1_1_crt_1_1_mqtt5_1_1_un_sub_ack_packet.html) returned by the broker; the promise is rejected with an error if anything goes wrong before the [UnsubAckPacket](https://aws.github.io/aws-iot-device-sdk-cpp-v2/class_aws_1_1_crt_1_1_mqtt5_1_1_un_sub_ack_packet.html) is received. You should always check the reason codes of a [UnsubAckPacket](https://aws.github.io/aws-iot-device-sdk-cpp-v2/class_aws_1_1_crt_1_1_mqtt5_1_1_un_sub_ack_packet.html) completion to determine if the unsubscribe operation actually succeeded.
```

Comment:
One of the links leads to java docs. Also, there are other places in the guide with Java links, I won't mark everyone of them.